module/modmanager/manager.go

```diff
@@ -633,11 +633,14 @@ func (mgr *Manager) ResolveImplicitDependenciesInConfig(ctx context.Context, con
 			// continue if this component is not being provided by a module.
 			continue
 		}
-		if !slices.ContainsFunc(conf.Modified.Modules, func(elem config.Module) bool {
+
+		//nolint:gocritic
```

Comment:
what is the linter complaining about here?

Reply:
linter wants me to do `conf.Modified.Modules = append(conf.Modified.Modules, conf.Added.Modules...)` but we don't want to modify the actual slice, just make a new local one

Reply:
hm, so `append` is a little weird and there are some cases where [it can actually update the underlying array](https://trstringer.com/golang-append/). 

Not exactly sure if that can happen here, but since we use both `conf.Modified.Modules` and `deltaModules` later on, it's safer to make a new array, copy `conf.Modified.Modules`, and then append `conf.Added.Modules`.

(Or once we upgrade to go1.22 we can just use https://pkg.go.dev/slices#Concat)